pkg/apis/build/v1alpha1/build_pod.go

```diff
@@ -165,16 +164,19 @@ func (b *Build) BuildPod(images BuildPodImages, secrets []corev1.Secret, taints
 
 	var cacheArgs []string
 	var cacheVolumes []corev1.VolumeMount
-	if b.Spec.CacheName == "" || config.OS == "windows" {
+	if b.Spec.Cache.ImageTag == "" && b.Spec.Cache.VolumeName == "" || config.OS == "windows" {
```

Comment:
The `home` volume with registry creds will need to be mounted in the restore step now to provide access to the registry cache.